src/test/java/Problem3Test.java

```diff
@@ -6,12 +6,19 @@
 public class Problem3Test {
     @Test
     public void catchTheBugInBook() {
-        // quiz
+        Book g = new BookFiction("t1", "vc2", "g5");
+        Book gr = new BookRomance("t1", "vc3");
+        assertFalse(g.equals(gr));
```

Comment:
the test logic is not correct. this test should still pass with the original bug. 

it should be the same id but other part should be different.